tests/test_cli.py

```diff
@@ -472,6 +472,55 @@ def test_import_tmuxinator(cli_args, inputs, tmpdir, monkeypatch):
         assert tmpdir.join('la.yaml').check()
 
 
+@pytest.mark.parametrize(
+    "cli_args,inputs",
+    [
+        (['freeze', 'mysession'], ['\n', 'y\n', './la.yaml\n', './la.yaml\n', 'y\n']),
+        (  # Exists
+            ['freeze', 'mysession'],
+            [
+                '\n',
+                'y\n',
+                './exists.yaml\n',
+                './exists.yaml\n',
+                './la.yaml\n',
+                './la.yaml\n',
+                'y\n',
+            ],
+        ),
+        (  # Imply current session if not entered
+            ['freeze'],
+            ['\n', 'y\n', './la.yaml\n', './la.yaml\n', 'y\n'],
+        ),
+        (
+            ['freeze'],
+            [
+                '\n',
+                'y\n',
+                './exists.yaml\n',
+                './exists.yaml\n',
+                './la.yaml\n',
+                './la.yaml\n',
+                'y\n',
+            ],
+        ),  # Exists
+    ],
+)
+def test_freeze(server, cli_args, inputs, tmpdir, monkeypatch):
+    monkeypatch.setenv('HOME', str(tmpdir))
+    tmpdir.join('exists.yaml').ensure()
+
+    server.new_session(session_name='mysession')
+
+    with tmpdir.as_cwd():
+        runner = CliRunner()
+        # Use tmux server (socket name) used in the test
+        cli_args = cli_args + ['-L', server.socket_name]
```

Comment:
@kstilwell 

`server` is a fixture for a randomly generated tmux session. In order to prevent it clobbering from any of the developer's sessions, it's ran on a custom socket name (via `-L socket-name`).

technically it's also possible tmuxp tests could be run in parallel: tmuxp / libtmux respects socket name and socket path

I created an issue at #495 to streamline freeze a bit more (not require entering the filename a second time)